### DISCUSSION TITLE
Detekt ignores semicolons in Kotlin code

### DISCUSSION THREAD

**Discussion by d35h** [detekt/detekt#3869] [Q&A] [answered]
Hello everyone,

I have a question, does anyone encounter the issue with detekt that it ignores semicolons in Kotlin code?

I have the following code:
```
@SpringBootApplication
class MyApp;

fun main(args: Array<String>) {
    runApplication<MyApp>(*args);
};
```

and when I run `./gradlew detekt` it fails on the SpreadOperator issue, but completely ignores semicolons.

I tried to explicitly feed detekt my config.yaml with NoSemicolons rule set to true, but the behavior is still the same.

I would appreciate any ideas!

Thank you,
Dani

**Comment by d35h** [comment, marked as answer]
Consider it solved, the following solved the issue:
```
    val detektVersion = "1.16.0-RC2"
    detekt("io.gitlab.arturbosch.detekt:detekt-formatting:$detektVersion")
    detekt("io.gitlab.arturbosch.detekt:detekt-cli:$detektVersion")
```

**Reply by cortinico** [reply]
Just a side note: you're using a older version of detekt (`1.16.0-RC2`) that was also a release candidate. You should consider moving to `1.17.1`.

**Reply by d35h** [reply]
@cortinico totally, but it was not the source of the issue, thanks though :)
